src/components/Calendar.tsx

```diff
@@ -1,85 +1,83 @@
-  { date: "2022-01-06", isCurrentMonth: true },
-  { date: "2022-01-07", isCurrentMonth: true },
-  { date: "2022-01-08", isCurrentMonth: true },
-  { date: "2022-01-09", isCurrentMonth: true },
-  { date: "2022-01-10", isCurrentMonth: true },
-  { date: "2022-01-11", isCurrentMonth: true },
-  { date: "2022-01-12", isCurrentMonth: true, isToday: true },
-  { date: "2022-01-13", isCurrentMonth: true },
-  { date: "2022-01-14", isCurrentMonth: true },
-  { date: "2022-01-15", isCurrentMonth: true },
-  { date: "2022-01-16", isCurrentMonth: true },
-  { date: "2022-01-17", isCurrentMonth: true },
-  { date: "2022-01-18", isCurrentMonth: true },
-  { date: "2022-01-19", isCurrentMonth: true },
-  { date: "2022-01-20", isCurrentMonth: true },
-  { date: "2022-01-21", isCurrentMonth: true },
-  { date: "2022-01-22", isCurrentMonth: true, isSelected: true },
-  { date: "2022-01-23", isCurrentMonth: true },
-  { date: "2022-01-24", isCurrentMonth: true },
-  { date: "2022-01-25", isCurrentMonth: true },
-  { date: "2022-01-26", isCurrentMonth: true },
-  { date: "2022-01-27", isCurrentMonth: true },
-  { date: "2022-01-28", isCurrentMonth: true },
-  { date: "2022-01-29", isCurrentMonth: true },
-  { date: "2022-01-30", isCurrentMonth: true },
-  { date: "2022-01-31", isCurrentMonth: true },
-  { date: "2022-02-01" },
-  { date: "2022-02-02" },
-  { date: "2022-02-03" },
-  { date: "2022-02-04" },
-  { date: "2022-02-05" },
-  { date: "2022-02-06" },
-];
+import { getToday, getCurrentMonth, getTithi } from "../helper/dates";
+import nepaliNumber from "../helper/nepaliNumber";
+import { Dispatch, SetStateAction, useState } from "react";
+import mahina from "../constants/mahina";
+import { Day } from "../types";
 
 function classNames(...classes: Array<string | undefined | boolean>) {
   return classes.filter(Boolean).join(" ");
 }
 
-export default function Calendar() {
-  const [activeDate, setActiveDate] = useState<string>("2022-1-1");
-  const [dayOfWeek, setDayOfWeek] = useState(1); //
-  console.log(activeDate.split("-"));
-  console.log(nepaliNumber(activeDate.split("-")[0]));
+interface YearData {
+  [key: string]: Day[];
+}
+interface Calender {
+  yearData: YearData | null;
+  setCurrentYear: Dispatch<SetStateAction<number>>;
+}
+
+const getMonthData = (yearData: YearData, currentMonth: number): Day[] => {
+  if (!yearData) return [];
+  const today = getToday().date;
+  const monthData = yearData[currentMonth + 1 < 10 ? "0" + (currentMonth + 1) : currentMonth + 1];
+  if (currentMonth === getToday().month) {
+    monthData[today - 1].is_today = true;
+  }
+  console.log(monthData);
+  return monthData;
+};
+
+export default function Calendar({ yearData, setCurrentYear }: Calender) {
+  const [currentMonth, setCurrentMonth] = useState(getCurrentMonth());
+  const [selectedDay, setSelectedDay] = useState<string>(
+    getCurrentMonth() === currentMonth ? getToday().dateStr : "01"
+  );
+
+  const handleNextMonth = () => {
```

Comment:
handle going beyond the range of our data, for example 2083 BS

Reply:
Should I toast an alert or anything specific to inform user that we don't have tha data beyond this ?

Reply:
No. Disable the prev/next button. Also, Could you add year information at the top? Allow users to click on it and select available years and month from the list as well.

Reply:
If you need to hardcode the limit of the available years, please use a config file or something for that kind of things. Will help us debug in the future.